cclib/parser/molcasparser.py

```diff
@@ -725,6 +738,62 @@ def extract(self, inputfile, line):
                     self.ccenergies= []
                 self.ccenergies.append(ccenergies)
 
+        #  ++    Primitive basis info:
+        #        ---------------------
+        #
+        #
+        #                      *****************************************************
+        #                      ******** Primitive Basis Functions (Valence) ********
+        #                      *****************************************************
+        #
+        #
+        #   Basis set:C.AUG-CC-PVQZ.........                                                          
+        #
+        #                    Type         
+        #                     s
+        #             No.      Exponent    Contraction Coefficients
+        #             1  0.339800000D+05   0.000091  -0.000019   0.000000   0.000000   0.000000   0.000000
+        #             2  0.508900000D+04   0.000704  -0.000151   0.000000   0.000000   0.000000   0.000000
+        # ...
+        # ...
+        #             29  0.424000000D+00   0.000000   1.000000
+        #
+        #   Number of primitives                                   93
+        #   Number of basis functions                              80
+        #
+        #  --
+        if '++    Primitive basis info:' in line:
+            self.skip_lines(inputfile, ['d', 'b', 'b', 's', 'header', 's', 'b'])
+            line = next(inputfile)
+            self.gbasis_array = []
+            while '--' not in line and '****' not in line:
+                if 'Basis set:' in line:
+                    basis_element = line.split()[1].split('.')[0].split(':')[1]
```

Comment:
Lots of splits going on there. What about a regex instead? Something like `\s+Basis set:([A-Za-z]{1,2})\.*.*`?

Reply:
Done in #632.